src/App.js

```diff
@@ -2,38 +2,57 @@ import React, { Component } from 'react';
 import movieData from './sample-data';
 import MovieContainer from './components/Movie-Container/Movie-Container'
 import SelectedMovie from './components/Selected-Movie/Selected-Movie';
-
+import Error from './components/Error/Error'
+import './App.css'
 class App extends Component {
   constructor() {
     super();
     this.state = {
       movies: [],
       movieSummary: false,
-      selectedMovie: ''
+      selectedMovie: '',
+      error: ''
     }
   }
 
-   handleClick = (movie) => {
-    this.setState({movieSummary: true, selectedMovie: movie})
+   handleClick = (id) => {
+     fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${id}`)
+     .then(data => data.json())
+     .then(data => this.setState(prevState => ({movies: prevState.movies, movieSummary: true, selectedMovie: data.movie})))
   }
 
     goHome = () => {
       this.setState({ movieSummary: false, selectedMovie: '' })
     }
+
+
   componentDidMount = () => {
-    this.setState({ movies: movieData.movies })
-  }
+    fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies').then((data) => {
+      if(data.ok) {
+        return data.json()
+      } else {
+        throw Error(data.error)
+      }
+    })
+    .then((response) => {
+
+      this.setState({movies: response.movies, selectedMovie: '', movieSummary: false})
+    })
+    .catch((err) => {
+      console.log(err)
+      })
+    }
 
   render() {
+    const activeError = this.state.error
     const movieSummary = this.state.movieSummary
     return (
-      
+
       <main className='Movie-Home-Page'>
         <nav className="nav">
           <button onClick={() => this.goHome()}>HOME</button>
```

Comment:
You might not have this logic anymore after adding Router, but if you do, you can simplify things a little bit by using a function reference since `goHome` doesn't take in any arguments: `onClick={this.goHome}`